src/main/java/io/stargate/sgv2/jsonapi/api/model/command/clause/sort/SortClause.java

```diff
@@ -93,6 +93,13 @@ public void validate(CollectionSchemaObject collection) {
         throw ErrorCodeV1.UNINDEXED_SORT_PATH.toApiException(
             "sort path '%s' is not indexed", sortExpression.path());
       }
+      if (!(DocumentConstants.Fields.VECTOR_EMBEDDING_FIELD.equals(sortExpression.path())
+              || DocumentConstants.Fields.VECTOR_EMBEDDING_TEXT_FIELD.equals(sortExpression.path()))
+          && sortExpression.vector() != null) {
+        throw ErrorCodeV1.INVALID_SORT_CLAUSE.toApiException(
+            "Vector sort for collection can be done only using `%s` field, field name used: `%s`  ",
```

Comment:
Is this correct ?  looks like it is saying we can only sort on $vector for collections but we also allow sort on $vectorize

Reply:
Want to say you can use only `$vector` field if sort is based on embedding vector as input. Yes, `$vectorize` does ANN sort but for end user it's based on text. Changed the error message, let me know.